File: src/eqn/ast/EqnAstNodeDouble.java
package eqn.ast;

public class EqnAstNodeDouble extends EqnAstNode {

    private final double valueOfNode;

    public EqnAstNodeDouble(String value) {
        super(value, Type.Constant);
        valueOfNode = Double.parseDouble(value);
    }

    public EqnAstNodeDouble(Double value) {
        super(Double.toString(value), Type.Constant);
        valueOfNode = value;
    }

    @Override
    public String toString() {
        return Double.toString(valueOfNode);
    }

    @Override
    public double evaluate() {
        return valueOfNode;
    }

    @Override
    public EqnAstNode simplify() {
        return this;
    }


}

File: src/eqn/ast/EqnAstNodeDouble.kt
package eqn.ast

import eqn.ast.base.EqnAstNode

class EqnAstNodeDouble : EqnAstNode {
    private var valueOfNode: Double

    override fun getConstantValue(): Double {
        throw UnsupportedOperationException("getConstantValue in EqnAstNodeDouble not supported")
    }

    constructor(value: String) : super(value, Type.Constant, PrecedenceType.Terminal) {
        valueOfNode = value.toDouble()
    }

    constructor(value: Double) : super(value.toString(), Type.Constant, PrecedenceType.Terminal) {
        valueOfNode = value
    }

    override fun toString(): String {
        return valueOfNode.toString()
    }

    override fun evaluate(arguments: Map<String, Double>?): Double {
        return valueOfNode
    }

    override fun arity(): Int = 0

    override fun simplify(arguments: Map<String, Double>?): EqnAstNode {
        return this
    }
}